File: pm_core_1/src/infrastructure/store/model/issue_type_scheme_item_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type IssueTypeSchemeItemModel struct {
	BaseModel
	TenantID    int64 `gorm:"column:tenant_id;not null" json:"tenantId"`
	SchemeID    int64 `gorm:"column:scheme_id;not null" json:"schemeId"`
	IssueTypeID int64 `gorm:"column:issue_type_id;not null" json:"issueTypeId"`
	Sequence    int   `gorm:"column:sequence;not null" json:"sequence"`
}

func (IssueTypeSchemeItemModel) TableName() string {
	return "issue_type_scheme_items"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/IssueTypeSchemeItemModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "issue_type_scheme_items")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class IssueTypeSchemeItemModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "scheme_id", nullable = false)
    private Long schemeId;

    @Column(name = "issue_type_id", nullable = false)
    private Long issueTypeId;

    @Column(name = "sequence")
    private Integer sequence;
}
